spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java

```diff
@@ -42,4 +42,9 @@ private SparkSQLProperties() {}
   // Controls whether to check the order of fields during writes
   public static final String CHECK_ORDERING = "spark.sql.iceberg.check-ordering";
   public static final boolean CHECK_ORDERING_DEFAULT = true;
+
+  // Controls whether to preserve the existing grouping of data while planning splits
+  public static final String PRESERVE_DATA_GROUPING =
+      "spark.sql.iceberg.split.preserve-data-grouping";
```

Comment:
Do we want "split" here? Would `preserve-data-grouping` be unclear? If not, what about `planning`?

Reply:
I was inspired by `read.split` table properties we have. If omitted, it is not clear whether it applies only to reads.
I like `planning`. We could also go for `read.split` to follow table properties but either one is equally good for me.

Reply:
Switched to `planning`.